src/app/hooks/useSort.js

```diff
@@ -32,13 +32,11 @@ export const useSort = (items, config = null) => {
         return sortableItems;
     }, [items, sortConfig]);
 
-    const requestSort = (activeSort = null) => {
-        let {key, direction} = activeSort
-        if (sortConfig && sortConfig.key === key && sortConfig.direction === "ASC") {
-            direction = "DESC";
-        }
-
-        setSortConfig({ key, direction });
+    const requestSort = key => {
+        setSortConfig(prevState => {
+            if (prevState.direction === "ASC") return { key, direction: "DESC" };
+            if (prevState.direction === "DESC") return { key, direction: "ASC" };
```

Comment:
Potentially we don't need this second if statement? That way we would also cover the case where we mistakenly pass a wrong direction value.

```suggestion
            return { key, direction: "ASC" };
```

Reply:
nope this needs to stay as the direction changes and manages the arrows selections